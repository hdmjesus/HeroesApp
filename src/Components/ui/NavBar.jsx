import React, { useContext } from "react";
import { Link, NavLink, useHistory } from "react-router-dom";
import { AuthContext } from "../../Context/AuthContext";
import { types } from "../../types/types";

export const NavBar = () => {
  const { user, dispatch } = useContext(AuthContext);
  const history = useHistory();

  const handleLogout = () => {
    //
    const userLogin = {};
    const actionLogout = { type: types.logout, payload: userLogin };
    dispatch(actionLogout);
    history.replace("/login");
  };
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark px-5">
      <Link className="navbar-brand" to="/">
        HeroesApp
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">
          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/marvel"
          >
            Marvel
          </NavLink>

          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/dc"
          >
            DC
          </NavLink>
          <NavLink
            activeClassName="active"
            className="nav-item nav-link"
            exact
            to="/search"
          >
            Search
          </NavLink>
        </div>
      </div>

      <div className="navbar-collapse collapse w-100 order-3 dual-collapse2">
        <ul className="navbar-nav ml-auto">
          <span className="nav-item nav-link text-info">{user.name}</span>
          <button className="btn text-white " onClick={handleLogout}>
            Logout
          </button>
        </ul>
      </div>
    </nav>
  );
};
