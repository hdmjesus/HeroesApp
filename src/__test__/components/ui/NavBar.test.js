import { mount } from "enzyme";
import { MemoryRouter } from "react-router";
import { Router } from "react-router-dom";
import { NavBar } from "../../../Components/ui/NavBar";
import { AuthContext } from "../../../Context/AuthContext";
import { types } from "../../../types/types";

describe("Pruebas en <NavBar/>", () => {
  const { Provider } = AuthContext;
  const historyMock = {
    push: jest.fn(),
    location: {},
    replace: jest.fn(),
    listen: jest.fn(),
    createHref: jest.fn(),
  };
  const contextValue = {
    dispatch: jest.fn(),
    user: { logged: true, name: "Jesus Hernandez" || expect.any(String) },
  };
  const wrapper = mount(
    <Provider value={contextValue}>
      <MemoryRouter>
        <Router history={historyMock}>
          <NavBar />
        </Router>
      </MemoryRouter>
    </Provider>
  );

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("should show good ", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text().trim()).toBe("Jesus Hernandez");
  });

  test("should call logout and use History", () => {
    wrapper.find("button").simulate("click");
    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.logout,
      payload: {},
    });

    expect(historyMock.replace).toHaveBeenCalledWith("/login");
  });
});
