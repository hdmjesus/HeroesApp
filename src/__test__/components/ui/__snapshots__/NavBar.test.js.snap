// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NavBar/> should show good  1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark px-5"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    HeroesApp
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="navbar-collapse collapse w-100 order-3 dual-collapse2"
  >
    <ul
      className="navbar-nav ml-auto"
    >
      <span
        className="nav-item nav-link text-info"
      >
        Jesus Hernandez
      </span>
      <button
        className="btn text-white "
        onClick={[Function]}
      >
        Logout
      </button>
    </ul>
  </div>
</nav>
`;
